pkg/deployer/lib/controller.go

```diff
@@ -381,9 +378,10 @@ func (c *controller) removeTestReconcileAnnotation(ctx context.Context, di *lsv1
 	logger, ctx := logging.FromContextOrNew(ctx, []interface{}{lc.KeyReconciledResource, client.ObjectKeyFromObject(di).String()})
 
 	logger.Info("remove test-reconcile annotation")
+	old := di.DeepCopy()
 	delete(di.Annotations, lsv1alpha1.OperationAnnotation)
-	if err := c.Writer().UpdateDeployItem(ctx, read_write_layer.W000149, di); client.IgnoreNotFound(err) != nil {
-		return lserrors.NewWrappedError(err, "RemoveTestReconcileAnnotation", "UpdateDeployItem", err.Error())
+	if err := c.Writer().PatchDeployItem(ctx, read_write_layer.W000149, di, client.MergeFrom(old)); client.IgnoreNotFound(err) != nil {
```

Comment:
I would not switch to a patch. One fundamental concept of the overall reconciliation is that we are only working with optimistic locking which is a fundamental concept we should not break anywhere. The logic is too complex to introduce such special stuff again. Especially for a test case.